{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "053c0118-b973-4900-a357-4d508098a4d9",
   "metadata": {},
   "source": [
    "## Run Workflow using Step Decorators\n",
    "\n",
    "The code and notebook in this directory shows how we can create a complete pipeline with step decorators.\n",
    "Each step of the pipeline is shown under the same run in MLFlow."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc6539d0-c10f-4b75-949e-4a125a7d0980",
   "metadata": {},
   "source": [
    "Lets restore the variables from the `00-start-here` notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a780a112-657c-4442-9681-2de9cc601a5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "%store -r \n",
    "\n",
    "%store\n",
    "\n",
    "try:\n",
    "    initialized\n",
    "except NameError:    \n",
    "    print(\"[ERROR] YOU HAVE TO RUN 00-start-here notebook   \")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a96eb89-b308-4745-8da1-93776e39bada",
   "metadata": {},
   "source": [
    "## Copy the Sagemaker distribution container to our private ECR repository"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "790d1417-c27e-4563-8060-56177684357f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import boto3\n",
    "import os\n",
    "\n",
    "ACCOUNT_ID = boto3.client('sts').get_caller_identity()['Account']\n",
    "REGION = boto3.session.Session().region_name\n",
    "\n",
    "REPO_NAME = f\"{project_prefix}-sagemaker-distribution-prod\"\n",
    "BASE_IMAGE=\"885854791233.dkr.ecr.us-east-1.amazonaws.com/sagemaker-distribution-prod@sha256:296c06cdf03dc6f1c3f1e7f8b4457f18178ab1b861ab485f33c64656d02d8799\"\n",
    "MY_REPO=f\"{ACCOUNT_ID}.dkr.ecr.{REGION}.amazonaws.com/{REPO_NAME}:latest\"\n",
    "\n",
    "\n",
    "os.environ[\"ACCOUNT_ID\"] = BASE_IMAGE\n",
    "os.environ[\"REGION\"] = REGION\n",
    "\n",
    "os.environ[\"REPO_NAME\"] = REPO_NAME\n",
    "os.environ[\"MY_REPO\"] = MY_REPO\n",
    "\n",
    "os.environ[\"BASE_IMAGE\"] = BASE_IMAGE\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91bb375a-f3aa-45fe-9709-9c8daed7dc40",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "REPO_NAME=$REPO_NAME\n",
    "\n",
    "# Check if the repository exists\n",
    "if aws ecr describe-repositories --repository-names \"$REPO_NAME\" > /dev/null 2>&1; then\n",
    "    echo \"Repository '$REPO_NAME' already exists.\"\n",
    "else\n",
    "    # Create the repository if it does not exist\n",
    "    aws ecr create-repository --repository-name \"$REPO_NAME\"\n",
    "    echo \"Repository '$REPO_NAME' created.\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7b7b206-98a4-45d8-af58-2f0157b7e479",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# download and push the image to our own image repository\n",
    "set -x\n",
    "\n",
    "docker pull \"$BASE_IMAGE\"\n",
    "aws ecr get-login-password --region \"$REGION\" | docker login --username AWS --password-stdin \"${ACCOUNT_ID}.dkr.ecr.${REGION}.amazonaws.com\"\n",
    "docker tag \"$BASE_IMAGE\" \"$MY_REPO\"\n",
    "docker push \"$MY_REPO\"\n",
    "\n",
    "echo \"Image pushed to ECR: $MY_REPO\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfd75289-b6c0-452e-871b-973ffcb55f35",
   "metadata": {},
   "source": [
    "## Run the pipeline locally"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a01d7e95-186a-4aee-ad93-be6223b79ae9",
   "metadata": {},
   "source": [
    "Let's first install the dependencies required to run this code locally"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f7e4b6c-f8c9-4a7a-b24f-2b3751203913",
   "metadata": {},
   "outputs": [],
   "source": [
    "%pip install -r requirements.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ecc88b1-fffe-49f4-8db6-6ba813ad555d",
   "metadata": {},
   "source": [
    "We create a config which will be used by default for each step. \n",
    "* `S3RootUri`: S3 location that will be used by default for the pipeline artifacts\n",
    "* `ImageUri`: Container image that will be used by default for each step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b20f6233-a82a-48a3-b9bf-76312b5b9f4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "config_yaml = f\"\"\"\n",
    "SchemaVersion: '1.0'\n",
    "SageMaker:\n",
    "  PythonSDK:\n",
    "    Modules:\n",
    "      RemoteFunction:\n",
    "        # role arn is not required if in SageMaker Notebook instance or SageMaker Studio\n",
    "        # Uncomment the following line and replace with the right execution role if in a local IDE\n",
    "        # RoleArn: <replace the role arn here>\n",
    "        S3RootUri: s3://{bucket_prefix}\n",
    "        ImageUri: {MY_REPO}\n",
    "        InstanceType: ml.m5.xlarge\n",
    "        Dependencies: ./requirements.txt\n",
    "        IncludeLocalWorkDir: true\n",
    "        PreExecutionCommands:\n",
    "        - \"sudo chmod -R 777 /opt/ml/model\"\n",
    "        CustomFileFilter:\n",
    "          IgnoreNamePatterns:\n",
    "          - \"data/*\"\n",
    "          - \"models/*\"\n",
    "          - \"*.ipynb\"\n",
    "          - \"__pycache__\"\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "print(config_yaml, file=open('config.yaml', 'w'))\n",
    "print(config_yaml)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8bb585f-6f84-43d4-9fa7-d222e432fc98",
   "metadata": {},
   "source": [
    "Now we run the pipeline in local mode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "21ab9de8-a6fc-423b-b924-696498a42da1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:33,468 sagemaker.remote_function INFO     Installing collected packages: aniso8601, querystring-parser, Mako, gunicorn, graphql-core, fsspec, deprecated, scikit-learn, opentelemetry-api, graphql-relay, alembic, opentelemetry-semantic-conventions, graphene, s3fs, opentelemetry-sdk, sagemaker, mlflow, sagemaker-mlflow\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:33,820 sagemaker.remote_function INFO       Attempting uninstall: fsspec\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:33,822 sagemaker.remote_function INFO         Found existing installation: fsspec 2023.6.0\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:33,828 sagemaker.remote_function INFO         Uninstalling fsspec-2023.6.0:\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:33,857 sagemaker.remote_function INFO           Successfully uninstalled fsspec-2023.6.0\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:34,004 sagemaker.remote_function INFO       Attempting uninstall: scikit-learn\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:34,007 sagemaker.remote_function INFO         Found existing installation: scikit-learn 1.4.2\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:34,090 sagemaker.remote_function INFO         Uninstalling scikit-learn-1.4.2:\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:34,343 sagemaker.remote_function INFO           Successfully uninstalled scikit-learn-1.4.2\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:36,654 sagemaker.remote_function INFO       Attempting uninstall: sagemaker\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:36,657 sagemaker.remote_function INFO         Found existing installation: sagemaker 2.227.0\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:36,707 sagemaker.remote_function INFO         Uninstalling sagemaker-2.227.0:\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:36,891 sagemaker.remote_function INFO           Successfully uninstalled sagemaker-2.227.0\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,089 sagemaker.remote_function INFO     Successfully installed Mako-1.3.5 alembic-1.13.3 aniso8601-9.0.1 deprecated-1.2.14 fsspec-2023.6.0 graphene-3.3 graphql-core-3.2.4 graphql-relay-3.2.0 gunicorn-22.0.0 mlflow-2.13.2 opentelemetry-api-1.27.0 opentelemetry-sdk-1.27.0 opentelemetry-semantic-conventions-0.48b0 querystring-parser-1.2.4 s3fs-2024.9.0 sagemaker-2.219.0 sagemaker-mlflow-0.1.0 scikit-learn-1.3.2\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,420 sagemaker.remote_function WARNING  WARNING: The directory '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,420 sagemaker.remote_function ERROR    ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,420 sagemaker.remote_function WARNING  autogluon-core 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,420 sagemaker.remote_function WARNING  autogluon-features 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,420 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,421 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pytorch-lightning<1.10.0,>=1.9.0, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,421 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,421 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchmetrics<0.12.0,>=0.11.0, but you have torchmetrics 1.0.3 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchvision<0.15.0, but you have torchvision 0.15.2a0+ab7b3e6 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  autogluon-tabular 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pytorch-lightning<1.10.0,>=1.7.4, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  datasets 2.20.0 requires fsspec[http]<=2024.5.0,>=2023.1.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function WARNING  jupyter-scheduler 2.7.1 requires fsspec==2023.6.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:39,422 sagemaker.remote_function INFO     Command /opt/conda/bin/python -m pip install -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt -U ran successfully\n",
      "cssvq8shw6-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "cssvq8shw6-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "cssvq8shw6-sagemaker-local  | INFO: Changing workspace to sagemaker_remote_function_workspace.\n",
      "cssvq8shw6-sagemaker-local  | INFO: No conda env provided. Invoking remote function\n",
      "cssvq8shw6-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "cssvq8shw6-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:42,193 sagemaker.remote_function INFO     Deserializing function code from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Evaluate/2024-10-04-13-41-38-525/function\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:43,374 sagemaker.remote_function INFO     Deserializing function arguments from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Evaluate/2024-10-04-13-41-38-525/arguments\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:43,542 sagemaker.remote_function INFO     Resolving pipeline variables\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:43,672 sagemaker.remote_function INFO     Invoking the function\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | 2024/10/04 13:42:44 INFO mlflow.tracking.fluent: Autologging successfully enabled for xgboost.\n",
      "cssvq8shw6-sagemaker-local  | 2024/10/04 13:42:44 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.\n",
      "cssvq8shw6-sagemaker-local  | ===Metrics for Test Set===\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | Predictions  0.0  ...   All\n",
      "cssvq8shw6-sagemaker-local  | Actuals           ...      \n",
      "cssvq8shw6-sagemaker-local  | 0            961  ...   966\n",
      "cssvq8shw6-sagemaker-local  | 1             10  ...    34\n",
      "cssvq8shw6-sagemaker-local  | All          971  ...  1000\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | [3 rows x 3 columns]\n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | \n",
      "cssvq8shw6-sagemaker-local  | Accuracy Model A: 98.50%\n",
      "cssvq8shw6-sagemaker-local  | Precision Model A: 0.83\n",
      "cssvq8shw6-sagemaker-local  | Recall Model A: 0.71\n",
      "cssvq8shw6-sagemaker-local  | AUC A: 0.94\n",
      "cssvq8shw6-sagemaker-local  | evaluation report: {'binary_classification_metrics': {'recall': {'value': 0.7058823529411765, 'standard_deviation': ''}, 'precision': {'value': 0.8275862068965517, 'standard_deviation': ''}, 'accuracy': {'value': 0.985, 'standard_deviation': ''}, 'auc': {'value': 0.9368225551090001, 'standard_deviation': ''}}}\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:44,870 sagemaker.remote_function INFO     Serializing the function return and uploading to s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Evaluate/results\n",
      "cssvq8shw6-sagemaker-local  | 2024-10-04 13:42:44,963 sagemaker.remote_function INFO     JSON Serializing the function return and uploading to s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Evaluate/results\n",
      "\u001b[Kcssvq8shw6-sagemaker-local exited with code 0\n",
      "Aborting on container exit...\n",
      " Container cssvq8shw6-sagemaker-local  Stopping\n",
      " Container cssvq8shw6-sagemaker-local  Stopped\n",
      "INFO:root:creating /home/sagemaker-user/tmp/tmpztsc0j32/artifacts/output/data\n",
      "INFO:sagemaker.local.image:===== Job Complete =====\n",
      "INFO:sagemaker.local.entities:Pipeline step 'Evaluate' SUCCEEDED.\n",
      "INFO:sagemaker.local.entities:Starting pipeline step: 'Register'\n",
      "INFO:sagemaker.local.image:'Docker Compose' found using Docker CLI.\n",
      "INFO:sagemaker.local.local_session:Starting training job\n",
      "WARNING:sagemaker.local.image:Using the short-lived AWS credentials found in session. They might expire while running.\n",
      "INFO:sagemaker.local.image:docker compose file: \n",
      "services:\n",
      "  sagemaker-local:\n",
      "    container_name: npm7ibitri-sagemaker-local\n",
      "    entrypoint:\n",
      "    - /bin/bash\n",
      "    - /opt/ml/input/data/sagemaker_remote_function_bootstrap/job_driver.sh\n",
      "    - --s3_base_uri\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline\n",
      "    - --region\n",
      "    - us-east-1\n",
      "    - --client_python_version\n",
      "    - '3.10'\n",
      "    - --client_sagemaker_pysdk_version\n",
      "    - 2.219.0\n",
      "    - --dependency_settings\n",
      "    - '{\"dependency_file\": \"requirements.txt\"}'\n",
      "    - --pipeline_step_name\n",
      "    - Register\n",
      "    - --pipeline_execution_id\n",
      "    - 44e548bc-62dc-43ea-bbba-504a17790cae\n",
      "    - --func_step_s3_dir\n",
      "    - 2024-10-04-13-41-38-525\n",
      "    - --property_references\n",
      "    - Execution.PipelineExecutionId\n",
      "    - 44e548bc-62dc-43ea-bbba-504a17790cae\n",
      "    - Steps.Preprocess.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Preprocess/results\n",
      "    - Steps.Train.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Train/results\n",
      "    - Steps.Evaluate.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Evaluate/results\n",
      "    - Steps.Preprocess.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Preprocess/results\n",
      "    - --serialize_output_to_json\n",
      "    - 'true'\n",
      "    environment:\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    image: 626723862963.dkr.ecr.us-east-1.amazonaws.com/amzn-sagemaker-distribution-prod:latest\n",
      "    network_mode: sagemaker\n",
      "    stdin_open: true\n",
      "    tty: true\n",
      "    volumes:\n",
      "    - /home/sagemaker-user/tmp/tmphojw1dop/sagemaker-local/output/data:/opt/ml/output/data\n",
      "    - /home/sagemaker-user/tmp/tmphojw1dop/sagemaker-local/output:/opt/ml/output\n",
      "    - /home/sagemaker-user/tmp/tmphojw1dop/sagemaker-local/input:/opt/ml/input\n",
      "    - /home/sagemaker-user/tmp/tmphojw1dop/model:/opt/ml/model\n",
      "    - /opt/ml/metadata:/opt/ml/metadata\n",
      "    - /home/sagemaker-user/tmp/tmplec8genh:/opt/ml/input/data/sagemaker_remote_function_bootstrap\n",
      "    - /home/sagemaker-user/tmp/tmpf466tfns:/opt/ml/input/data/pre_exec_script_and_dependencies\n",
      "    - /home/sagemaker-user/tmp/tmpeo01w4om:/opt/ml/input/data/2024-10-04-13-41-38-525\n",
      "version: '2.3'\n",
      "\n",
      "INFO:sagemaker.local.image:docker command: docker compose -f /home/sagemaker-user/tmp/tmphojw1dop/docker-compose.yaml up --build --abort-on-container-exit\n",
      "time=\"2024-10-04T13:42:47Z\" level=warning msg=\"/home/sagemaker-user/tmp/tmphojw1dop/docker-compose.yaml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion\"\n",
      " Container npm7ibitri-sagemaker-local  Creating\n",
      " Container npm7ibitri-sagemaker-local  Created\n",
      "Attaching to npm7ibitri-sagemaker-local\n",
      "npm7ibitri-sagemaker-local  | INFO: CONDA_PKGS_DIRS is set to '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/conda/pkgs'\n",
      "npm7ibitri-sagemaker-local  | INFO: PIP_CACHE_DIR is set to '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/pip'\n",
      "npm7ibitri-sagemaker-local  | INFO: Bootstraping runtime environment.\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,294 sagemaker.remote_function INFO     The job is running on non-root user: sagemaker-user. Adding write permissions to the following job output directories: ['/opt/ml/output', '/opt/ml/model', '/tmp'].\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,294 sagemaker.remote_function INFO     Executing 'sudo chmod -R 777 /opt/ml/output /opt/ml/model /tmp'.\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,307 sagemaker.remote_function INFO     Successfully unpacked workspace archive at '/home/sagemaker-user'.\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,308 sagemaker.remote_function INFO     Running pre-execution commands in '/home/sagemaker-user/sagemaker_remote_function_workspace/pre_exec.sh'\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,316 sagemaker.remote_function INFO     Running command: '/opt/conda/bin/python -m pip install -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt -U' in the dir: '/home/sagemaker-user' \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,871 sagemaker.remote_function INFO     Collecting sagemaker==2.219.0 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:47,895 sagemaker.remote_function INFO       Downloading sagemaker-2.219.0-py3-none-any.whl.metadata (14 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,002 sagemaker.remote_function INFO     Collecting scikit-learn==1.3.2 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,007 sagemaker.remote_function INFO       Downloading scikit_learn-1.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,039 sagemaker.remote_function INFO     Collecting s3fs (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,044 sagemaker.remote_function INFO       Downloading s3fs-2024.9.0-py3-none-any.whl.metadata (1.6 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,085 sagemaker.remote_function INFO     Collecting mlflow==2.13.2 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,091 sagemaker.remote_function INFO       Downloading mlflow-2.13.2-py3-none-any.whl.metadata (29 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,102 sagemaker.remote_function INFO     Collecting sagemaker-mlflow (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 5))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,107 sagemaker.remote_function INFO       Downloading sagemaker_mlflow-0.1.0-py3-none-any.whl.metadata (3.3 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,170 sagemaker.remote_function INFO     Requirement already satisfied: attrs<24,>=23.1.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (23.2.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,171 sagemaker.remote_function INFO     Requirement already satisfied: boto3<2.0,>=1.33.3 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.34.131)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,172 sagemaker.remote_function INFO     Requirement already satisfied: cloudpickle==2.2.1 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.2.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,173 sagemaker.remote_function INFO     Requirement already satisfied: google-pasta in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.2.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,174 sagemaker.remote_function INFO     Requirement already satisfied: numpy<2.0,>=1.9.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.26.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,175 sagemaker.remote_function INFO     Requirement already satisfied: protobuf<5.0,>=3.12 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.24.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,176 sagemaker.remote_function INFO     Requirement already satisfied: smdebug-rulesconfig==1.0.1 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.0.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,176 sagemaker.remote_function INFO     Requirement already satisfied: importlib-metadata<7.0,>=1.4.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (6.10.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,177 sagemaker.remote_function INFO     Requirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (24.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,178 sagemaker.remote_function INFO     Requirement already satisfied: pandas in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.1.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,179 sagemaker.remote_function INFO     Requirement already satisfied: pathos in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,179 sagemaker.remote_function INFO     Requirement already satisfied: schema in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.7.7)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,181 sagemaker.remote_function INFO     Requirement already satisfied: PyYAML~=6.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (6.0.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,181 sagemaker.remote_function INFO     Requirement already satisfied: jsonschema in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.17.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,182 sagemaker.remote_function INFO     Requirement already satisfied: platformdirs in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.2.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,183 sagemaker.remote_function INFO     Requirement already satisfied: tblib<4,>=1.7.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.0.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,184 sagemaker.remote_function INFO     Requirement already satisfied: urllib3<3.0.0,>=1.26.8 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.26.19)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,184 sagemaker.remote_function INFO     Requirement already satisfied: requests in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.32.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,185 sagemaker.remote_function INFO     Requirement already satisfied: docker in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (7.1.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,185 sagemaker.remote_function INFO     Requirement already satisfied: tqdm in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.66.5)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,186 sagemaker.remote_function INFO     Requirement already satisfied: psutil in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (5.9.8)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,221 sagemaker.remote_function INFO     Requirement already satisfied: scipy>=1.5.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (1.11.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,222 sagemaker.remote_function INFO     Requirement already satisfied: joblib>=1.1.1 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (1.4.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,222 sagemaker.remote_function INFO     Requirement already satisfied: threadpoolctl>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (3.5.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,303 sagemaker.remote_function INFO     Requirement already satisfied: Flask<4 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,350 sagemaker.remote_function INFO     Collecting alembic!=1.10.0,<2 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,354 sagemaker.remote_function INFO       Downloading alembic-1.13.3-py3-none-any.whl.metadata (7.4 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,356 sagemaker.remote_function INFO     Requirement already satisfied: cachetools<6,>=5.0.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (5.4.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,357 sagemaker.remote_function INFO     Requirement already satisfied: click<9,>=7.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (8.1.7)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,358 sagemaker.remote_function INFO     Requirement already satisfied: entrypoints<1 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,359 sagemaker.remote_function INFO     Requirement already satisfied: gitpython<4,>=3.1.9 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.43)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,381 sagemaker.remote_function INFO     Collecting graphene<4 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,385 sagemaker.remote_function INFO       Downloading graphene-3.3-py2.py3-none-any.whl.metadata (7.7 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,387 sagemaker.remote_function INFO     Requirement already satisfied: markdown<4,>=3.3 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.6)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,388 sagemaker.remote_function INFO     Requirement already satisfied: matplotlib<4 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.9.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,416 sagemaker.remote_function INFO     Collecting opentelemetry-api<3,>=1.0.0 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,420 sagemaker.remote_function INFO       Downloading opentelemetry_api-1.27.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,448 sagemaker.remote_function INFO     Collecting opentelemetry-sdk<3,>=1.0.0 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,452 sagemaker.remote_function INFO       Downloading opentelemetry_sdk-1.27.0-py3-none-any.whl.metadata (1.5 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,454 sagemaker.remote_function INFO     Requirement already satisfied: pyarrow<16,>=4.0.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (15.0.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,455 sagemaker.remote_function INFO     Requirement already satisfied: pytz<2025 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2023.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,465 sagemaker.remote_function INFO     Collecting querystring-parser<2 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,469 sagemaker.remote_function INFO       Downloading querystring_parser-1.2.4-py2.py3-none-any.whl.metadata (559 bytes)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,475 sagemaker.remote_function INFO     Requirement already satisfied: sqlalchemy<3,>=1.4.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.0.30)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,477 sagemaker.remote_function INFO     Requirement already satisfied: sqlparse<1,>=0.4.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.5.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,479 sagemaker.remote_function INFO     Requirement already satisfied: Jinja2<4,>=2.11 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,511 sagemaker.remote_function INFO     Collecting gunicorn<23 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,516 sagemaker.remote_function INFO       Downloading gunicorn-22.0.0-py3-none-any.whl.metadata (4.4 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,527 sagemaker.remote_function INFO     Requirement already satisfied: aiobotocore<3.0.0,>=2.5.4 in /opt/conda/lib/python3.10/site-packages (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (2.13.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,558 sagemaker.remote_function INFO     Collecting fsspec==2024.9.0.* (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,563 sagemaker.remote_function INFO       Downloading fsspec-2024.9.0-py3-none-any.whl.metadata (11 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,565 sagemaker.remote_function INFO     Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /opt/conda/lib/python3.10/site-packages (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (3.9.5)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,684 sagemaker.remote_function INFO     Requirement already satisfied: botocore<1.34.132,>=1.34.70 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.34.131)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,685 sagemaker.remote_function INFO     Requirement already satisfied: wrapt<2.0.0,>=1.10.10 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.14.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,686 sagemaker.remote_function INFO     Requirement already satisfied: aioitertools<1.0.0,>=0.5.1 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (0.11.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,698 sagemaker.remote_function INFO     Requirement already satisfied: aiosignal>=1.1.2 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.3.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,700 sagemaker.remote_function INFO     Requirement already satisfied: frozenlist>=1.1.1 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.4.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,701 sagemaker.remote_function INFO     Requirement already satisfied: multidict<7.0,>=4.5 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (6.0.5)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,701 sagemaker.remote_function INFO     Requirement already satisfied: yarl<2.0,>=1.0 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.9.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,703 sagemaker.remote_function INFO     Requirement already satisfied: async-timeout<5.0,>=4.0 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (4.0.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,739 sagemaker.remote_function INFO     Collecting Mako (from alembic!=1.10.0,<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,744 sagemaker.remote_function INFO       Downloading Mako-1.3.5-py3-none-any.whl.metadata (2.9 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,746 sagemaker.remote_function INFO     Requirement already satisfied: typing-extensions>=4 in /opt/conda/lib/python3.10/site-packages (from alembic!=1.10.0,<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.12.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,775 sagemaker.remote_function INFO     Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /opt/conda/lib/python3.10/site-packages (from boto3<2.0,>=1.33.3->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.0.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,776 sagemaker.remote_function INFO     Requirement already satisfied: s3transfer<0.11.0,>=0.10.0 in /opt/conda/lib/python3.10/site-packages (from boto3<2.0,>=1.33.3->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.10.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,806 sagemaker.remote_function INFO     Requirement already satisfied: Werkzeug>=3.0.0 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,807 sagemaker.remote_function INFO     Requirement already satisfied: itsdangerous>=2.1.2 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.2.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,808 sagemaker.remote_function INFO     Requirement already satisfied: blinker>=1.6.2 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.8.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,829 sagemaker.remote_function INFO     Requirement already satisfied: gitdb<5,>=4.0.1 in /opt/conda/lib/python3.10/site-packages (from gitpython<4,>=3.1.9->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.0.11)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,885 sagemaker.remote_function INFO     Collecting graphql-core<3.3,>=3.1 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,891 sagemaker.remote_function INFO       Downloading graphql_core-3.2.4-py3-none-any.whl.metadata (10 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,914 sagemaker.remote_function INFO     Collecting graphql-relay<3.3,>=3.1 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,919 sagemaker.remote_function INFO       Downloading graphql_relay-3.2.0-py3-none-any.whl.metadata (12 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,939 sagemaker.remote_function INFO     Collecting aniso8601<10,>=8 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:48,943 sagemaker.remote_function INFO       Downloading aniso8601-9.0.1-py2.py3-none-any.whl.metadata (23 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,007 sagemaker.remote_function INFO     Requirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.10/site-packages (from importlib-metadata<7.0,>=1.4.0->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.19.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,016 sagemaker.remote_function INFO     Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.10/site-packages (from Jinja2<4,>=2.11->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.1.5)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,045 sagemaker.remote_function INFO     Requirement already satisfied: contourpy>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.2.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,046 sagemaker.remote_function INFO     Requirement already satisfied: cycler>=0.10 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.12.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,047 sagemaker.remote_function INFO     Requirement already satisfied: fonttools>=4.22.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.53.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,048 sagemaker.remote_function INFO     Requirement already satisfied: kiwisolver>=1.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.4.5)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,049 sagemaker.remote_function INFO     Requirement already satisfied: pillow>=8 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (10.4.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,050 sagemaker.remote_function INFO     Requirement already satisfied: pyparsing>=2.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,055 sagemaker.remote_function INFO     Requirement already satisfied: python-dateutil>=2.7 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.9.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,086 sagemaker.remote_function INFO     Collecting deprecated>=1.2.6 (from opentelemetry-api<3,>=1.0.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,090 sagemaker.remote_function INFO       Downloading Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,125 sagemaker.remote_function INFO     Collecting opentelemetry-semantic-conventions==0.48b0 (from opentelemetry-sdk<3,>=1.0.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,129 sagemaker.remote_function INFO       Downloading opentelemetry_semantic_conventions-0.48b0-py3-none-any.whl.metadata (2.4 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,228 sagemaker.remote_function INFO     Requirement already satisfied: tzdata>=2022.1 in /opt/conda/lib/python3.10/site-packages (from pandas->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2024.1)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,256 sagemaker.remote_function INFO     Requirement already satisfied: six in /opt/conda/lib/python3.10/site-packages (from querystring-parser<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.16.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,269 sagemaker.remote_function INFO     Requirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.3.2)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,270 sagemaker.remote_function INFO     Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.7)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,272 sagemaker.remote_function INFO     Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2024.7.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,380 sagemaker.remote_function INFO     Requirement already satisfied: greenlet!=0.4.17 in /opt/conda/lib/python3.10/site-packages (from sqlalchemy<3,>=1.4.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,463 sagemaker.remote_function INFO     Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /opt/conda/lib/python3.10/site-packages (from jsonschema->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.20.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,475 sagemaker.remote_function INFO     Requirement already satisfied: ppft>=1.7.6.8 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.7.6.8)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,476 sagemaker.remote_function INFO     Requirement already satisfied: dill>=0.3.8 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.8)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,477 sagemaker.remote_function INFO     Requirement already satisfied: pox>=0.3.4 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.4)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,478 sagemaker.remote_function INFO     Requirement already satisfied: multiprocess>=0.70.16 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.70.16)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,693 sagemaker.remote_function INFO     Requirement already satisfied: smmap<6,>=3.0.1 in /opt/conda/lib/python3.10/site-packages (from gitdb<5,>=4.0.1->gitpython<4,>=3.1.9->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (5.0.0)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,971 sagemaker.remote_function INFO     Downloading sagemaker-2.219.0-py3-none-any.whl (1.5 MB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,983 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 156.3 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:49,990 sagemaker.remote_function INFO     Downloading scikit_learn-1.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.8 MB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,033 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.8/10.8 MB 270.3 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,037 sagemaker.remote_function INFO     Downloading mlflow-2.13.2-py3-none-any.whl (25.0 MB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,123 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 25.0/25.0 MB 310.2 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,127 sagemaker.remote_function INFO     Downloading s3fs-2024.9.0-py3-none-any.whl (29 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,132 sagemaker.remote_function INFO     Downloading fsspec-2024.9.0-py3-none-any.whl (179 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,136 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.3/179.3 kB 358.3 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,140 sagemaker.remote_function INFO     Downloading sagemaker_mlflow-0.1.0-py3-none-any.whl (24 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,144 sagemaker.remote_function INFO     Downloading alembic-1.13.3-py3-none-any.whl (233 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,147 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 233.2/233.2 kB 385.1 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,151 sagemaker.remote_function INFO     Downloading graphene-3.3-py2.py3-none-any.whl (128 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,153 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 kB 317.4 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,157 sagemaker.remote_function INFO     Downloading gunicorn-22.0.0-py3-none-any.whl (84 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,159 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.4/84.4 kB 328.2 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,163 sagemaker.remote_function INFO     Downloading opentelemetry_api-1.27.0-py3-none-any.whl (63 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,166 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.0/64.0 kB 293.9 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,170 sagemaker.remote_function INFO     Downloading opentelemetry_sdk-1.27.0-py3-none-any.whl (110 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,173 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 110.5/110.5 kB 323.0 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,177 sagemaker.remote_function INFO     Downloading opentelemetry_semantic_conventions-0.48b0-py3-none-any.whl (149 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,180 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.7/149.7 kB 289.4 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,183 sagemaker.remote_function INFO     Downloading querystring_parser-1.2.4-py2.py3-none-any.whl (7.9 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,187 sagemaker.remote_function INFO     Downloading aniso8601-9.0.1-py2.py3-none-any.whl (52 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,189 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.8/52.8 kB 220.8 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,193 sagemaker.remote_function INFO     Downloading Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,197 sagemaker.remote_function INFO     Downloading graphql_core-3.2.4-py3-none-any.whl (203 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,200 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 203.2/203.2 kB 320.8 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,204 sagemaker.remote_function INFO     Downloading graphql_relay-3.2.0-py3-none-any.whl (16 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,207 sagemaker.remote_function INFO     Downloading Mako-1.3.5-py3-none-any.whl (78 kB)\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:50,210 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.6/78.6 kB 321.2 MB/s eta 0:00:00\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:53,856 sagemaker.remote_function INFO     Installing collected packages: aniso8601, querystring-parser, Mako, gunicorn, graphql-core, fsspec, deprecated, scikit-learn, opentelemetry-api, graphql-relay, alembic, opentelemetry-semantic-conventions, graphene, s3fs, opentelemetry-sdk, sagemaker, mlflow, sagemaker-mlflow\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,212 sagemaker.remote_function INFO       Attempting uninstall: fsspec\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,215 sagemaker.remote_function INFO         Found existing installation: fsspec 2023.6.0\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,221 sagemaker.remote_function INFO         Uninstalling fsspec-2023.6.0:\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,246 sagemaker.remote_function INFO           Successfully uninstalled fsspec-2023.6.0\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,389 sagemaker.remote_function INFO       Attempting uninstall: scikit-learn\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,392 sagemaker.remote_function INFO         Found existing installation: scikit-learn 1.4.2\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,458 sagemaker.remote_function INFO         Uninstalling scikit-learn-1.4.2:\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:54,673 sagemaker.remote_function INFO           Successfully uninstalled scikit-learn-1.4.2\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:57,025 sagemaker.remote_function INFO       Attempting uninstall: sagemaker\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:57,030 sagemaker.remote_function INFO         Found existing installation: sagemaker 2.227.0\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:57,083 sagemaker.remote_function INFO         Uninstalling sagemaker-2.227.0:\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:57,249 sagemaker.remote_function INFO           Successfully uninstalled sagemaker-2.227.0\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,410 sagemaker.remote_function INFO     Successfully installed Mako-1.3.5 alembic-1.13.3 aniso8601-9.0.1 deprecated-1.2.14 fsspec-2023.6.0 graphene-3.3 graphql-core-3.2.4 graphql-relay-3.2.0 gunicorn-22.0.0 mlflow-2.13.2 opentelemetry-api-1.27.0 opentelemetry-sdk-1.27.0 opentelemetry-semantic-conventions-0.48b0 querystring-parser-1.2.4 s3fs-2024.9.0 sagemaker-2.219.0 sagemaker-mlflow-0.1.0 scikit-learn-1.3.2\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  WARNING: The directory '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function ERROR    ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-core 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-features 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pytorch-lightning<1.10.0,>=1.9.0, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchmetrics<0.12.0,>=0.11.0, but you have torchmetrics 1.0.3 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchvision<0.15.0, but you have torchvision 0.15.2a0+ab7b3e6 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-tabular 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pytorch-lightning<1.10.0,>=1.7.4, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  datasets 2.20.0 requires fsspec[http]<=2024.5.0,>=2023.1.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,763 sagemaker.remote_function WARNING  jupyter-scheduler 2.7.1 requires fsspec==2023.6.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "npm7ibitri-sagemaker-local  | \n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:42:59,764 sagemaker.remote_function INFO     Command /opt/conda/bin/python -m pip install -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt -U ran successfully\n",
      "npm7ibitri-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "npm7ibitri-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "npm7ibitri-sagemaker-local  | INFO: Changing workspace to sagemaker_remote_function_workspace.\n",
      "npm7ibitri-sagemaker-local  | INFO: No conda env provided. Invoking remote function\n",
      "npm7ibitri-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "npm7ibitri-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:02,663 sagemaker.remote_function INFO     Deserializing function code from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Register/2024-10-04-13-41-38-525/function\n",
      "npm7ibitri-sagemaker-local  | /opt/conda/lib/python3.10/site-packages/paramiko/pkey.py:100: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.\n",
      "npm7ibitri-sagemaker-local  |   \"cipher\": algorithms.TripleDES,\n",
      "npm7ibitri-sagemaker-local  | /opt/conda/lib/python3.10/site-packages/paramiko/transport.py:259: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.\n",
      "npm7ibitri-sagemaker-local  |   \"class\": algorithms.TripleDES,\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:04,059 sagemaker.remote_function INFO     Deserializing function arguments from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Register/2024-10-04-13-41-38-525/arguments\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:04,166 sagemaker.remote_function INFO     Resolving pipeline variables\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:04,348 sagemaker.remote_function INFO     Invoking the function\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | 2024/10/04 13:43:05 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.\n",
      "npm7ibitri-sagemaker-local  | 2024/10/04 13:43:05 INFO mlflow.tracking.fluent: Autologging successfully enabled for xgboost.\n",
      "npm7ibitri-sagemaker-local  | INFO:sagemaker.image_uris:Defaulting to only available Python version: py3\n",
      "npm7ibitri-sagemaker-local  | INFO:sagemaker.image_uris:Defaulting to only supported image scope: cpu.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Either inference spec or model is provided. ModelBuilder is not handling MLflow model input\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Skipping auto detection as the image uri is provided 683313688378.dkr.ecr.us-east-1.amazonaws.com/sagemaker-scikit-learn:1.2-1-cpu-py3\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Packaging provided requirements.txt from requirements_inference.txt\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Uploading the model resources to bucket=sagemaker-us-east-1-626723862963, key_prefix=sagemaker-scikit-learn-2024-10-04-13-43-05-708.\n",
      "Uploading model artifacts: 100%|████████████████████████| 12114/12114 [00:00<00:00, 62814.23bytes/s]\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Model resources uploaded to: s3://sagemaker-us-east-1-626723862963/sagemaker-scikit-learn-2024-10-04-13-43-05-708/serve.tar.gz\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Either inference spec or model is provided. ModelBuilder is not handling MLflow model input\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Auto detect container url for the provided model and on instance ml.c5.xlarge\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Autodetecting image since image_uri was not provided in ModelBuilder()\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: WARNING:     Auto detection is only supported for single models DLCs with a framework backend.\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Autodetected framework is xgboost\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Autodetected framework version is 1.7.6\n",
      "npm7ibitri-sagemaker-local  | INFO:sagemaker.image_uris:Ignoring unnecessary Python version: py310.\n",
      "npm7ibitri-sagemaker-local  | INFO:sagemaker.image_uris:Ignoring unnecessary instance type: ml.c5.xlarge.\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: INFO:     Auto detected 683313688378.dkr.ecr.us-east-1.amazonaws.com/sagemaker-xgboost:1.7-1. Proceeding with the the deployment.\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Packaging provided requirements.txt from requirements_inference.txt\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Uploading the model resources to bucket=sagemaker-us-east-1-626723862963, key_prefix=sagemaker-xgboost-2024-10-04-13-43-06-773.\n",
      "Uploading model artifacts: 100%|███████████████████████| 53731/53731 [00:00<00:00, 373024.34bytes/s]\n",
      "npm7ibitri-sagemaker-local  | ModelBuilder: DEBUG:     Model resources uploaded to: s3://sagemaker-us-east-1-626723862963/sagemaker-xgboost-2024-10-04-13-43-06-773/serve.tar.gz\n",
      "npm7ibitri-sagemaker-local  | INFO:aiobotocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | Successfully registered model package arn:aws:sagemaker:us-east-1:626723862963:model-package/amzn-sm-btd-model-package-group/4.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:08,228 sagemaker.remote_function INFO     Serializing the function return and uploading to s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Register/results\n",
      "npm7ibitri-sagemaker-local  | 2024-10-04 13:43:08,285 sagemaker.remote_function INFO     JSON Serializing the function return and uploading to s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Register/results\n",
      "\u001b[Knpm7ibitri-sagemaker-local exited with code 0\n",
      "Aborting on container exit...\n",
      " Container npm7ibitri-sagemaker-local  Stopping\n",
      " Container npm7ibitri-sagemaker-local  Stopped\n",
      "INFO:root:creating /home/sagemaker-user/tmp/tmphojw1dop/artifacts/output/data\n",
      "INFO:sagemaker.local.image:===== Job Complete =====\n",
      "INFO:sagemaker.local.entities:Pipeline step 'Register' SUCCEEDED.\n",
      "INFO:sagemaker.local.entities:Starting pipeline step: 'Deploy'\n",
      "INFO:sagemaker.local.image:'Docker Compose' found using Docker CLI.\n",
      "INFO:sagemaker.local.local_session:Starting training job\n",
      "WARNING:sagemaker.local.image:Using the short-lived AWS credentials found in session. They might expire while running.\n",
      "INFO:sagemaker.local.image:docker compose file: \n",
      "services:\n",
      "  sagemaker-local:\n",
      "    container_name: 1vuiptfd7v-sagemaker-local\n",
      "    entrypoint:\n",
      "    - /bin/bash\n",
      "    - /opt/ml/input/data/sagemaker_remote_function_bootstrap/job_driver.sh\n",
      "    - --s3_base_uri\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline\n",
      "    - --region\n",
      "    - us-east-1\n",
      "    - --client_python_version\n",
      "    - '3.10'\n",
      "    - --client_sagemaker_pysdk_version\n",
      "    - 2.219.0\n",
      "    - --dependency_settings\n",
      "    - '{\"dependency_file\": \"requirements.txt\"}'\n",
      "    - --pipeline_step_name\n",
      "    - Deploy\n",
      "    - --pipeline_execution_id\n",
      "    - 44e548bc-62dc-43ea-bbba-504a17790cae\n",
      "    - --func_step_s3_dir\n",
      "    - 2024-10-04-13-41-38-525\n",
      "    - --property_references\n",
      "    - Execution.PipelineExecutionId\n",
      "    - 44e548bc-62dc-43ea-bbba-504a17790cae\n",
      "    - Steps.Register.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Register/results\n",
      "    - Parameters.deploy_model\n",
      "    - 'True'\n",
      "    - Steps.Preprocess.OutputDataConfig.S3OutputPath\n",
      "    - s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Preprocess/results\n",
      "    environment:\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    - '[Masked]'\n",
      "    image: 626723862963.dkr.ecr.us-east-1.amazonaws.com/amzn-sagemaker-distribution-prod:latest\n",
      "    network_mode: sagemaker\n",
      "    stdin_open: true\n",
      "    tty: true\n",
      "    volumes:\n",
      "    - /home/sagemaker-user/tmp/tmp8pvy6o4j/sagemaker-local/output/data:/opt/ml/output/data\n",
      "    - /home/sagemaker-user/tmp/tmp8pvy6o4j/sagemaker-local/output:/opt/ml/output\n",
      "    - /home/sagemaker-user/tmp/tmp8pvy6o4j/sagemaker-local/input:/opt/ml/input\n",
      "    - /home/sagemaker-user/tmp/tmp8pvy6o4j/model:/opt/ml/model\n",
      "    - /opt/ml/metadata:/opt/ml/metadata\n",
      "    - /home/sagemaker-user/tmp/tmp7b6exarb:/opt/ml/input/data/sagemaker_remote_function_bootstrap\n",
      "    - /home/sagemaker-user/tmp/tmp3x86geso:/opt/ml/input/data/pre_exec_script_and_dependencies\n",
      "    - /home/sagemaker-user/tmp/tmp2soztqzo:/opt/ml/input/data/2024-10-04-13-41-38-525\n",
      "version: '2.3'\n",
      "\n",
      "INFO:sagemaker.local.image:docker command: docker compose -f /home/sagemaker-user/tmp/tmp8pvy6o4j/docker-compose.yaml up --build --abort-on-container-exit\n",
      "time=\"2024-10-04T13:43:10Z\" level=warning msg=\"/home/sagemaker-user/tmp/tmp8pvy6o4j/docker-compose.yaml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion\"\n",
      " Container 1vuiptfd7v-sagemaker-local  Creating\n",
      " Container 1vuiptfd7v-sagemaker-local  Created\n",
      "Attaching to 1vuiptfd7v-sagemaker-local\n",
      "1vuiptfd7v-sagemaker-local  | INFO: CONDA_PKGS_DIRS is set to '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/conda/pkgs'\n",
      "1vuiptfd7v-sagemaker-local  | INFO: PIP_CACHE_DIR is set to '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/pip'\n",
      "1vuiptfd7v-sagemaker-local  | INFO: Bootstraping runtime environment.\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:10,707 sagemaker.remote_function INFO     The job is running on non-root user: sagemaker-user. Adding write permissions to the following job output directories: ['/opt/ml/output', '/opt/ml/model', '/tmp'].\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:10,707 sagemaker.remote_function INFO     Executing 'sudo chmod -R 777 /opt/ml/output /opt/ml/model /tmp'.\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:10,721 sagemaker.remote_function INFO     Successfully unpacked workspace archive at '/home/sagemaker-user'.\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:10,722 sagemaker.remote_function INFO     Running pre-execution commands in '/home/sagemaker-user/sagemaker_remote_function_workspace/pre_exec.sh'\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:10,731 sagemaker.remote_function INFO     Running command: '/opt/conda/bin/python -m pip install -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt -U' in the dir: '/home/sagemaker-user' \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,313 sagemaker.remote_function INFO     Collecting sagemaker==2.219.0 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,333 sagemaker.remote_function INFO       Downloading sagemaker-2.219.0-py3-none-any.whl.metadata (14 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,438 sagemaker.remote_function INFO     Collecting scikit-learn==1.3.2 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,444 sagemaker.remote_function INFO       Downloading scikit_learn-1.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,474 sagemaker.remote_function INFO     Collecting s3fs (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,478 sagemaker.remote_function INFO       Downloading s3fs-2024.9.0-py3-none-any.whl.metadata (1.6 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,520 sagemaker.remote_function INFO     Collecting mlflow==2.13.2 (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,524 sagemaker.remote_function INFO       Downloading mlflow-2.13.2-py3-none-any.whl.metadata (29 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,537 sagemaker.remote_function INFO     Collecting sagemaker-mlflow (from -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 5))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,540 sagemaker.remote_function INFO       Downloading sagemaker_mlflow-0.1.0-py3-none-any.whl.metadata (3.3 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,602 sagemaker.remote_function INFO     Requirement already satisfied: attrs<24,>=23.1.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (23.2.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,603 sagemaker.remote_function INFO     Requirement already satisfied: boto3<2.0,>=1.33.3 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.34.131)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,604 sagemaker.remote_function INFO     Requirement already satisfied: cloudpickle==2.2.1 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.2.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,605 sagemaker.remote_function INFO     Requirement already satisfied: google-pasta in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.2.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,606 sagemaker.remote_function INFO     Requirement already satisfied: numpy<2.0,>=1.9.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.26.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,607 sagemaker.remote_function INFO     Requirement already satisfied: protobuf<5.0,>=3.12 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.24.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,608 sagemaker.remote_function INFO     Requirement already satisfied: smdebug-rulesconfig==1.0.1 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.0.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,609 sagemaker.remote_function INFO     Requirement already satisfied: importlib-metadata<7.0,>=1.4.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (6.10.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,610 sagemaker.remote_function INFO     Requirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (24.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,610 sagemaker.remote_function INFO     Requirement already satisfied: pandas in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.1.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,611 sagemaker.remote_function INFO     Requirement already satisfied: pathos in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,611 sagemaker.remote_function INFO     Requirement already satisfied: schema in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.7.7)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,612 sagemaker.remote_function INFO     Requirement already satisfied: PyYAML~=6.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (6.0.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,613 sagemaker.remote_function INFO     Requirement already satisfied: jsonschema in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.17.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,613 sagemaker.remote_function INFO     Requirement already satisfied: platformdirs in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.2.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,614 sagemaker.remote_function INFO     Requirement already satisfied: tblib<4,>=1.7.0 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.0.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,615 sagemaker.remote_function INFO     Requirement already satisfied: urllib3<3.0.0,>=1.26.8 in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.26.19)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,616 sagemaker.remote_function INFO     Requirement already satisfied: requests in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2.32.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,616 sagemaker.remote_function INFO     Requirement already satisfied: docker in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (7.1.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,617 sagemaker.remote_function INFO     Requirement already satisfied: tqdm in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (4.66.5)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,618 sagemaker.remote_function INFO     Requirement already satisfied: psutil in /opt/conda/lib/python3.10/site-packages (from sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (5.9.8)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,651 sagemaker.remote_function INFO     Requirement already satisfied: scipy>=1.5.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (1.11.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,652 sagemaker.remote_function INFO     Requirement already satisfied: joblib>=1.1.1 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (1.4.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,654 sagemaker.remote_function INFO     Requirement already satisfied: threadpoolctl>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn==1.3.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 2)) (3.5.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,739 sagemaker.remote_function INFO     Requirement already satisfied: Flask<4 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,784 sagemaker.remote_function INFO     Collecting alembic!=1.10.0,<2 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,788 sagemaker.remote_function INFO       Downloading alembic-1.13.3-py3-none-any.whl.metadata (7.4 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,790 sagemaker.remote_function INFO     Requirement already satisfied: cachetools<6,>=5.0.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (5.4.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,791 sagemaker.remote_function INFO     Requirement already satisfied: click<9,>=7.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (8.1.7)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,792 sagemaker.remote_function INFO     Requirement already satisfied: entrypoints<1 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,793 sagemaker.remote_function INFO     Requirement already satisfied: gitpython<4,>=3.1.9 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.43)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,816 sagemaker.remote_function INFO     Collecting graphene<4 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,819 sagemaker.remote_function INFO       Downloading graphene-3.3-py2.py3-none-any.whl.metadata (7.7 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,821 sagemaker.remote_function INFO     Requirement already satisfied: markdown<4,>=3.3 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.6)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,822 sagemaker.remote_function INFO     Requirement already satisfied: matplotlib<4 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.9.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,852 sagemaker.remote_function INFO     Collecting opentelemetry-api<3,>=1.0.0 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,855 sagemaker.remote_function INFO       Downloading opentelemetry_api-1.27.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,884 sagemaker.remote_function INFO     Collecting opentelemetry-sdk<3,>=1.0.0 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,888 sagemaker.remote_function INFO       Downloading opentelemetry_sdk-1.27.0-py3-none-any.whl.metadata (1.5 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,894 sagemaker.remote_function INFO     Requirement already satisfied: pyarrow<16,>=4.0.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (15.0.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,896 sagemaker.remote_function INFO     Requirement already satisfied: pytz<2025 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2023.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,905 sagemaker.remote_function INFO     Collecting querystring-parser<2 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,909 sagemaker.remote_function INFO       Downloading querystring_parser-1.2.4-py2.py3-none-any.whl.metadata (559 bytes)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,914 sagemaker.remote_function INFO     Requirement already satisfied: sqlalchemy<3,>=1.4.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.0.30)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,915 sagemaker.remote_function INFO     Requirement already satisfied: sqlparse<1,>=0.4.0 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.5.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,917 sagemaker.remote_function INFO     Requirement already satisfied: Jinja2<4,>=2.11 in /opt/conda/lib/python3.10/site-packages (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,942 sagemaker.remote_function INFO     Collecting gunicorn<23 (from mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,945 sagemaker.remote_function INFO       Downloading gunicorn-22.0.0-py3-none-any.whl.metadata (4.4 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,956 sagemaker.remote_function INFO     Requirement already satisfied: aiobotocore<3.0.0,>=2.5.4 in /opt/conda/lib/python3.10/site-packages (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (2.13.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,987 sagemaker.remote_function INFO     Collecting fsspec==2024.9.0.* (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,992 sagemaker.remote_function INFO       Downloading fsspec-2024.9.0-py3-none-any.whl.metadata (11 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:11,995 sagemaker.remote_function INFO     Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /opt/conda/lib/python3.10/site-packages (from s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (3.9.5)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,124 sagemaker.remote_function INFO     Requirement already satisfied: botocore<1.34.132,>=1.34.70 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.34.131)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,126 sagemaker.remote_function INFO     Requirement already satisfied: wrapt<2.0.0,>=1.10.10 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.14.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,127 sagemaker.remote_function INFO     Requirement already satisfied: aioitertools<1.0.0,>=0.5.1 in /opt/conda/lib/python3.10/site-packages (from aiobotocore<3.0.0,>=2.5.4->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (0.11.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,139 sagemaker.remote_function INFO     Requirement already satisfied: aiosignal>=1.1.2 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.3.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,141 sagemaker.remote_function INFO     Requirement already satisfied: frozenlist>=1.1.1 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.4.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,142 sagemaker.remote_function INFO     Requirement already satisfied: multidict<7.0,>=4.5 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (6.0.5)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,143 sagemaker.remote_function INFO     Requirement already satisfied: yarl<2.0,>=1.0 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (1.9.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,144 sagemaker.remote_function INFO     Requirement already satisfied: async-timeout<5.0,>=4.0 in /opt/conda/lib/python3.10/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->s3fs->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 3)) (4.0.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,175 sagemaker.remote_function INFO     Collecting Mako (from alembic!=1.10.0,<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,179 sagemaker.remote_function INFO       Downloading Mako-1.3.5-py3-none-any.whl.metadata (2.9 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,180 sagemaker.remote_function INFO     Requirement already satisfied: typing-extensions>=4 in /opt/conda/lib/python3.10/site-packages (from alembic!=1.10.0,<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.12.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,211 sagemaker.remote_function INFO     Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /opt/conda/lib/python3.10/site-packages (from boto3<2.0,>=1.33.3->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.0.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,212 sagemaker.remote_function INFO     Requirement already satisfied: s3transfer<0.11.0,>=0.10.0 in /opt/conda/lib/python3.10/site-packages (from boto3<2.0,>=1.33.3->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.10.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,243 sagemaker.remote_function INFO     Requirement already satisfied: Werkzeug>=3.0.0 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,244 sagemaker.remote_function INFO     Requirement already satisfied: itsdangerous>=2.1.2 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.2.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,245 sagemaker.remote_function INFO     Requirement already satisfied: blinker>=1.6.2 in /opt/conda/lib/python3.10/site-packages (from Flask<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.8.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,267 sagemaker.remote_function INFO     Requirement already satisfied: gitdb<5,>=4.0.1 in /opt/conda/lib/python3.10/site-packages (from gitpython<4,>=3.1.9->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.0.11)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,326 sagemaker.remote_function INFO     Collecting graphql-core<3.3,>=3.1 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,330 sagemaker.remote_function INFO       Downloading graphql_core-3.2.4-py3-none-any.whl.metadata (10 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,346 sagemaker.remote_function INFO     Collecting graphql-relay<3.3,>=3.1 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,350 sagemaker.remote_function INFO       Downloading graphql_relay-3.2.0-py3-none-any.whl.metadata (12 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,366 sagemaker.remote_function INFO     Collecting aniso8601<10,>=8 (from graphene<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,369 sagemaker.remote_function INFO       Downloading aniso8601-9.0.1-py2.py3-none-any.whl.metadata (23 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,436 sagemaker.remote_function INFO     Requirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.10/site-packages (from importlib-metadata<7.0,>=1.4.0->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.19.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,443 sagemaker.remote_function INFO     Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.10/site-packages (from Jinja2<4,>=2.11->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.1.5)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,473 sagemaker.remote_function INFO     Requirement already satisfied: contourpy>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.2.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,474 sagemaker.remote_function INFO     Requirement already satisfied: cycler>=0.10 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (0.12.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,475 sagemaker.remote_function INFO     Requirement already satisfied: fonttools>=4.22.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (4.53.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,476 sagemaker.remote_function INFO     Requirement already satisfied: kiwisolver>=1.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.4.5)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,477 sagemaker.remote_function INFO     Requirement already satisfied: pillow>=8 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (10.4.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,479 sagemaker.remote_function INFO     Requirement already satisfied: pyparsing>=2.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.1.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,480 sagemaker.remote_function INFO     Requirement already satisfied: python-dateutil>=2.7 in /opt/conda/lib/python3.10/site-packages (from matplotlib<4->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (2.9.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,510 sagemaker.remote_function INFO     Collecting deprecated>=1.2.6 (from opentelemetry-api<3,>=1.0.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,514 sagemaker.remote_function INFO       Downloading Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,552 sagemaker.remote_function INFO     Collecting opentelemetry-semantic-conventions==0.48b0 (from opentelemetry-sdk<3,>=1.0.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4))\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,556 sagemaker.remote_function INFO       Downloading opentelemetry_semantic_conventions-0.48b0-py3-none-any.whl.metadata (2.4 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,648 sagemaker.remote_function INFO     Requirement already satisfied: tzdata>=2022.1 in /opt/conda/lib/python3.10/site-packages (from pandas->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2024.1)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,679 sagemaker.remote_function INFO     Requirement already satisfied: six in /opt/conda/lib/python3.10/site-packages (from querystring-parser<2->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (1.16.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,697 sagemaker.remote_function INFO     Requirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.3.2)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,698 sagemaker.remote_function INFO     Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (3.7)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,700 sagemaker.remote_function INFO     Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (2024.7.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,810 sagemaker.remote_function INFO     Requirement already satisfied: greenlet!=0.4.17 in /opt/conda/lib/python3.10/site-packages (from sqlalchemy<3,>=1.4.0->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (3.0.3)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,887 sagemaker.remote_function INFO     Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /opt/conda/lib/python3.10/site-packages (from jsonschema->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.20.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,896 sagemaker.remote_function INFO     Requirement already satisfied: ppft>=1.7.6.8 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (1.7.6.8)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,897 sagemaker.remote_function INFO     Requirement already satisfied: dill>=0.3.8 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.8)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,898 sagemaker.remote_function INFO     Requirement already satisfied: pox>=0.3.4 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.3.4)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:12,900 sagemaker.remote_function INFO     Requirement already satisfied: multiprocess>=0.70.16 in /opt/conda/lib/python3.10/site-packages (from pathos->sagemaker==2.219.0->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 1)) (0.70.16)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,140 sagemaker.remote_function INFO     Requirement already satisfied: smmap<6,>=3.0.1 in /opt/conda/lib/python3.10/site-packages (from gitdb<5,>=4.0.1->gitpython<4,>=3.1.9->mlflow==2.13.2->-r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt (line 4)) (5.0.0)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,423 sagemaker.remote_function INFO     Downloading sagemaker-2.219.0-py3-none-any.whl (1.5 MB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,433 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 201.0 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,437 sagemaker.remote_function INFO     Downloading scikit_learn-1.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.8 MB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,486 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.8/10.8 MB 231.0 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,493 sagemaker.remote_function INFO     Downloading mlflow-2.13.2-py3-none-any.whl (25.0 MB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,601 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 25.0/25.0 MB 209.3 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,605 sagemaker.remote_function INFO     Downloading s3fs-2024.9.0-py3-none-any.whl (29 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,608 sagemaker.remote_function INFO     Downloading fsspec-2024.9.0-py3-none-any.whl (179 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,612 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.3/179.3 kB 365.8 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,615 sagemaker.remote_function INFO     Downloading sagemaker_mlflow-0.1.0-py3-none-any.whl (24 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,619 sagemaker.remote_function INFO     Downloading alembic-1.13.3-py3-none-any.whl (233 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,622 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 233.2/233.2 kB 351.4 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,626 sagemaker.remote_function INFO     Downloading graphene-3.3-py2.py3-none-any.whl (128 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,629 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 kB 343.9 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,632 sagemaker.remote_function INFO     Downloading gunicorn-22.0.0-py3-none-any.whl (84 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,636 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.4/84.4 kB 315.5 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,639 sagemaker.remote_function INFO     Downloading opentelemetry_api-1.27.0-py3-none-any.whl (63 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,641 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.0/64.0 kB 325.8 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,644 sagemaker.remote_function INFO     Downloading opentelemetry_sdk-1.27.0-py3-none-any.whl (110 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,647 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 110.5/110.5 kB 340.9 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,650 sagemaker.remote_function INFO     Downloading opentelemetry_semantic_conventions-0.48b0-py3-none-any.whl (149 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,653 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.7/149.7 kB 355.6 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,656 sagemaker.remote_function INFO     Downloading querystring_parser-1.2.4-py2.py3-none-any.whl (7.9 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,660 sagemaker.remote_function INFO     Downloading aniso8601-9.0.1-py2.py3-none-any.whl (52 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,662 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.8/52.8 kB 245.6 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,665 sagemaker.remote_function INFO     Downloading Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,668 sagemaker.remote_function INFO     Downloading graphql_core-3.2.4-py3-none-any.whl (203 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,671 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 203.2/203.2 kB 372.3 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,675 sagemaker.remote_function INFO     Downloading graphql_relay-3.2.0-py3-none-any.whl (16 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,678 sagemaker.remote_function INFO     Downloading Mako-1.3.5-py3-none-any.whl (78 kB)\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:13,681 sagemaker.remote_function INFO        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.6/78.6 kB 210.6 MB/s eta 0:00:00\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,332 sagemaker.remote_function INFO     Installing collected packages: aniso8601, querystring-parser, Mako, gunicorn, graphql-core, fsspec, deprecated, scikit-learn, opentelemetry-api, graphql-relay, alembic, opentelemetry-semantic-conventions, graphene, s3fs, opentelemetry-sdk, sagemaker, mlflow, sagemaker-mlflow\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,684 sagemaker.remote_function INFO       Attempting uninstall: fsspec\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,686 sagemaker.remote_function INFO         Found existing installation: fsspec 2023.6.0\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,692 sagemaker.remote_function INFO         Uninstalling fsspec-2023.6.0:\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,719 sagemaker.remote_function INFO           Successfully uninstalled fsspec-2023.6.0\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,860 sagemaker.remote_function INFO       Attempting uninstall: scikit-learn\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,864 sagemaker.remote_function INFO         Found existing installation: scikit-learn 1.4.2\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:17,929 sagemaker.remote_function INFO         Uninstalling scikit-learn-1.4.2:\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:18,151 sagemaker.remote_function INFO           Successfully uninstalled scikit-learn-1.4.2\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:20,486 sagemaker.remote_function INFO       Attempting uninstall: sagemaker\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:20,489 sagemaker.remote_function INFO         Found existing installation: sagemaker 2.227.0\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:20,545 sagemaker.remote_function INFO         Uninstalling sagemaker-2.227.0:\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:20,715 sagemaker.remote_function INFO           Successfully uninstalled sagemaker-2.227.0\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:22,947 sagemaker.remote_function INFO     Successfully installed Mako-1.3.5 alembic-1.13.3 aniso8601-9.0.1 deprecated-1.2.14 fsspec-2023.6.0 graphene-3.3 graphql-core-3.2.4 graphql-relay-3.2.0 gunicorn-22.0.0 mlflow-2.13.2 opentelemetry-api-1.27.0 opentelemetry-sdk-1.27.0 opentelemetry-semantic-conventions-0.48b0 querystring-parser-1.2.4 s3fs-2024.9.0 sagemaker-2.219.0 sagemaker-mlflow-0.1.0 scikit-learn-1.3.2\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,285 sagemaker.remote_function WARNING  WARNING: The directory '/opt/ml/cache/sm_remotefunction_user_dependencies_cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,285 sagemaker.remote_function ERROR    ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,285 sagemaker.remote_function WARNING  autogluon-core 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,285 sagemaker.remote_function WARNING  autogluon-features 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,285 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires pytorch-lightning<1.10.0,>=1.9.0, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchmetrics<0.12.0,>=0.11.0, but you have torchmetrics 1.0.3 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-multimodal 0.8.3 requires torchvision<0.15.0, but you have torchvision 0.15.2a0+ab7b3e6 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-tabular 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pandas<1.6,>=1.4.1, but you have pandas 2.1.4 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires pytorch-lightning<1.10.0,>=1.7.4, but you have pytorch-lightning 2.0.9 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  autogluon-timeseries 0.8.3 requires torch<1.14,>=1.9, but you have torch 2.0.0.post104 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  datasets 2.20.0 requires fsspec[http]<=2024.5.0,>=2023.1.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function WARNING  jupyter-scheduler 2.7.1 requires fsspec==2023.6.0, but you have fsspec 2024.9.0 which is incompatible.\n",
      "1vuiptfd7v-sagemaker-local  | \n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:23,286 sagemaker.remote_function INFO     Command /opt/conda/bin/python -m pip install -r /home/sagemaker-user/sagemaker_remote_function_workspace/requirements.txt -U ran successfully\n",
      "1vuiptfd7v-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "1vuiptfd7v-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "1vuiptfd7v-sagemaker-local  | INFO: Changing workspace to sagemaker_remote_function_workspace.\n",
      "1vuiptfd7v-sagemaker-local  | INFO: No conda env provided. Invoking remote function\n",
      "1vuiptfd7v-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
      "1vuiptfd7v-sagemaker-local  | sagemaker.config INFO - Not applying SDK defaults from location: /home/sagemaker-user/.config/sagemaker/config.yaml\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:26,051 sagemaker.remote_function INFO     Deserializing function code from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Deploy/2024-10-04-13-41-38-525/function\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:26,792 sagemaker.remote_function INFO     Deserializing function arguments from s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/Deploy/2024-10-04-13-41-38-525/arguments\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:26,878 sagemaker.remote_function INFO     Resolving pipeline variables\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:27,327 sagemaker.remote_function INFO     Invoking the function\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | 2024/10/04 13:43:28 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.\n",
      "1vuiptfd7v-sagemaker-local  | Skipped deploy model step based on parameter configuration.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | INFO:botocore.credentials:Found credentials in environment variables.\n",
      "1vuiptfd7v-sagemaker-local  | 2024-10-04 13:43:28,408 sagemaker.remote_function INFO     Serializing the function return and uploading to s3://sagemaker-us-east-1-626723862963/amzn/amzn-sm-btd-pipeline/44e548bc-62dc-43ea-bbba-504a17790cae/Deploy/results\n",
      "\u001b[K1vuiptfd7v-sagemaker-local exited with code 0\n",
      "Aborting on container exit...\n",
      " Container 1vuiptfd7v-sagemaker-local  Stopping\n",
      " Container 1vuiptfd7v-sagemaker-local  Stopped\n",
      "INFO:root:creating /home/sagemaker-user/tmp/tmp8pvy6o4j/artifacts/output/data\n",
      "INFO:sagemaker.local.image:===== Job Complete =====\n",
      "INFO:sagemaker.local.entities:Pipeline step 'Deploy' SUCCEEDED.\n",
      "INFO:sagemaker.local.entities:Pipeline execution 44e548bc-62dc-43ea-bbba-504a17790cae SUCCEEDED\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ[\"MLFLOW_TRACKING_ARN\"] = mlflow_arn\n",
    "os.environ[\"LOCAL_MODE\"] = \"True\"\n",
    "!python pipeline.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88d7b3f9-3e06-4936-a655-d4ae73fc26fc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c93eac8b-dea6-4aff-8b7b-e36ab2272c50",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
